### PR TITLE
chore: change ml models location

### PR DESCRIPTION
**Pull request by eratostennis**
## Description

Change ml models location from google drive to AWS s3 in order to solve the following problem.
https://github.com/autowarefoundation/autoware/issues/200

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/lidar_apollo_instance_segmentation/package.xml [hunk 1]
```diff
@@ -11,6 +11,7 @@
   <author email="yukky.saito@gmail.com">Yukihiro Saito</author>
 
   <buildtool_depend>ament_cmake_auto</buildtool_depend>
+  <buildtool_depend>curl</buildtool_depend>
```
Comment (wep21) [inline perception/lidar_apollo_instance_segmentation/package.xml:14]:
```suggestion
  <buildtool_depend>libcurl_vendor</buildtool_depend>
```

File: perception/lidar_centerpoint/package.xml [hunk 2]
```diff
@@ -8,6 +8,7 @@
   <license>Apache License 2.0</license>
 
   <buildtool_depend>ament_cmake_auto</buildtool_depend>
+  <buildtool_depend>curl</buildtool_depend>
```
Comment (wep21) [inline perception/lidar_centerpoint/package.xml:11]:
```suggestion
  <buildtool_depend>libcurl_vendor</buildtool_depend>
```

File: perception/tensorrt_yolo/package.xml [hunk 3]
```diff
@@ -8,6 +8,7 @@
   <license>Apache License 2.0</license>
 
   <buildtool_depend>ament_cmake_auto</buildtool_depend>
+  <buildtool_depend>curl</buildtool_depend>
```
Comment (wep21) [inline perception/tensorrt_yolo/package.xml:11]:
```suggestion
  <buildtool_depend>libcurl_vendor</buildtool_depend>
```

File: perception/traffic_light_classifier/package.xml [hunk 4]
```diff
@@ -8,6 +8,7 @@
   <license>Apache License 2.0</license>
 
   <buildtool_depend>ament_cmake_auto</buildtool_depend>
+  <buildtool_depend>curl</buildtool_depend>
```
Comment (wep21) [inline perception/traffic_light_classifier/package.xml:11]:
```suggestion
  <buildtool_depend>libcurl_vendor</buildtool_depend>
```

File: perception/traffic_light_ssd_fine_detector/package.xml [hunk 5]
```diff
@@ -8,6 +8,7 @@
   <license>Apache License 2.0</license>
 
   <buildtool_depend>ament_cmake_auto</buildtool_depend>
+  <buildtool_depend>curl</buildtool_depend>
```
Comment (wep21) [inline perception/traffic_light_ssd_fine_detector/package.xml:11]:
```suggestion
  <buildtool_depend>libcurl_vendor</buildtool_depend>
```

**Comment by wep21** [comment]
It may be more simple to use file(DOWNLOAD).

**Comment by eratostennis** [comment]
> It may be more simple to use file(DOWNLOAD).

@wep21
I fixed it. Please review again.
Thanks.

File: perception/lidar_apollo_instance_segmentation/CMakeLists.txt [hunk 6]
```diff
@@ -65,35 +65,53 @@ else()
...
+    if(EXISTS ${FILE_PATH})
+      execute_process(COMMAND echo "found ${FILE_NAME}")
+      file(MD5 ${FILE_PATH} EXISTING_FILE_HASH)
+      if(${FILE_HASH} STREQUAL ${EXISTING_FILE_HASH})
+        execute_process(COMMAND echo "same ${FILE_NAME}")
+        message(STATUS "File already exists.")
+      else()
+        execute_process(COMMAND echo "diff ${FILE_NAME}")
+        message(STATUS "File hash changes. Downloading now ...")
+        file(DOWNLOAD https://awf.ml.dev.web.auto/perception/models/${FILE_NAME} ${FILE_PATH} STATUS DOWNLOAD_STATUS)
```
Comment (wep21) [inline perception/lidar_apollo_instance_segmentation/CMakeLists.txt:89]: I guess you can use this argument to check md5 hash.
```
Additional options to DOWNLOAD are:

EXPECTED_HASH ALGO=<value>

Verify that the downloaded content hash matches the expected value, where ALGO is one of the algorithms supported by file(<HASH>). If it does not match, the operation fails with an error. It is an error to specify this if DOWNLOAD is not given a <file>.

EXPECTED_MD5 <value>
Historical short-hand for EXPECTED_HASH MD5=<value>. It is an error to specify this if DOWNLOAD is not given a <file>.
```

File: perception/lidar_apollo_instance_segmentation/CMakeLists.txt [hunk 7]
```diff
@@ -65,35 +65,53 @@ else()
...
+  set(DATA_PATH "${CMAKE_CURRENT_SOURCE_DIR}/data")
+  if(NOT EXISTS "${DATA_PATH}")
+    execute_process(COMMAND mkdir -p ${DATA_PATH})
   endif()
 
+  function(download FILE_NAME FILE_HASH)
+    message(STATUS "Checking and downloading ${FILE_NAME}")
+    set(FILE_PATH ${DATA_PATH}/${FILE_NAME})
+    set(STATUS_CODE 0)
+    execute_process(COMMAND echo "start ${FILE_NAME}" 1>&2)
```
Comment (wep21) [inline perception/lidar_apollo_instance_segmentation/CMakeLists.txt:79]:
```suggestion
    message(STATUS "start ${FILE_NAME}")
```

File: perception/lidar_apollo_instance_segmentation/CMakeLists.txt [hunk 8]
```diff
@@ -65,35 +65,53 @@ else()
...
+    if(EXISTS ${FILE_PATH})
+      execute_process(COMMAND echo "found ${FILE_NAME}")
```
Comment (wep21) [inline perception/lidar_apollo_instance_segmentation/CMakeLists.txt:81]:
```suggestion
      message(STATUS "found ${FILE_NAME}")
```

File: perception/lidar_apollo_instance_segmentation/CMakeLists.txt [hunk 9]
```diff
@@ -65,35 +65,53 @@ else()
...
+      file(MD5 ${FILE_PATH} EXISTING_FILE_HASH)
+      if(${FILE_HASH} STREQUAL ${EXISTING_FILE_HASH})
+        execute_process(COMMAND echo "same ${FILE_NAME}")
```
Comment (wep21) [inline perception/lidar_apollo_instance_segmentation/CMakeLists.txt:84]:
```suggestion
        message(STATUS "same ${FILE_NAME}")
```

File: perception/lidar_centerpoint/CMakeLists.txt [hunk 10]
```diff
@@ -66,45 +66,55 @@ else()
...
   set(DATA_PATH ${CMAKE_CURRENT_SOURCE_DIR}/data)
   execute_process(COMMAND mkdir -p ${DATA_PATH})
 
-  function(download FILE_NAME GFILE_ID FILE_HASH)
-    # https://drive.google.com/file/d/GFILE_ID/view
+  function(download FILE_NAME FILE_HASH)
     message(STATUS "Checking and downloading ${FILE_NAME}")
     set(FILE_PATH ${DATA_PATH}/${FILE_NAME})
+    set(STATUS_CODE 0)
+    execute_process(COMMAND echo "start ${FILE_NAME}" 1>&2)
```
Comment (wep21) [inline perception/lidar_centerpoint/CMakeLists.txt:79]: Ditto

File: perception/tensorrt_yolo/CMakeLists.txt [hunk 11]
```diff
@@ -68,100 +68,54 @@ else()
...
-  )
+set(DATA_PATH "${CMAKE_CURRENT_SOURCE_DIR}/data")
+if(NOT EXISTS "${DATA_PATH}")
+  execute_process(COMMAND mkdir -p ${DATA_PATH})
 endif()
+function(download FILE_NAME FILE_HASH)
+  message(STATUS "Checking and downloading ${FILE_NAME}")
+  set(FILE_PATH ${DATA_PATH}/${FILE_NAME})
+  set(STATUS_CODE 0)
+  execute_process(COMMAND echo "start ${FILE_NAME}" 1>&2)
```
Comment (wep21) [inline perception/tensorrt_yolo/CMakeLists.txt:79]: Ditto

File: perception/traffic_light_classifier/CMakeLists.txt [hunk 12]
```diff
@@ -66,44 +66,47 @@ else()
...
-if(EXISTS "${FILE}")
-  file(MD5 "${FILE}" EXISTING_FILE_HASH)
-  if(NOT "${PRETRAINED_MODEL_HASH}" EQUAL "${EXISTING_FILE_HASH}")
-    message(STATUS "... file hash changed. Downloading now ...")
-    execute_process(COMMAND gdown --quiet "${PRETRAINED_MODEL_LINK}" -O ${PATH}/traffic_light_classifier_mobilenetv2.onnx)
+function(download FILE_NAME FILE_HASH)
+  message(STATUS "Checking and downloading ${FILE_NAME}")
+  set(FILE_PATH ${DATA_PATH}/${FILE_NAME})
+  set(STATUS_CODE 0)
+  execute_process(COMMAND echo "start ${FILE_NAME}" 1>&2)
```
Comment (wep21) [inline perception/traffic_light_classifier/CMakeLists.txt:77]: Ditto

File: perception/traffic_light_ssd_fine_detector/CMakeLists.txt [hunk 13]
```diff
@@ -68,45 +68,51 @@ else()
...
-if(EXISTS "${FILE}")
-  file(MD5 "${FILE}" EXISTING_FILE_HASH)
-  if(NOT "${PRETRAINED_MODEL_HASH}" EQUAL "${EXISTING_FILE_HASH}")
-    message(STATUS "... file hash changed. Downloading now ...")
-    execute_process(COMMAND gdown --quiet "${PRETRAINED_MODEL_LINK}" -O ${PATH}/mb2-ssd-lite-tlr.onnx)
+function(download FILE_NAME FILE_HASH)
+  message(STATUS "Checking and downloading ${FILE_NAME}")
+  set(FILE_PATH ${DATA_PATH}/${FILE_NAME})
+  set(STATUS_CODE 0)
+  execute_process(COMMAND echo "start ${FILE_NAME}" 1>&2)
```
Comment (wep21) [inline perception/traffic_light_ssd_fine_detector/CMakeLists.txt:83]: Ditto

File: perception/lidar_apollo_instance_segmentation/CMakeLists.txt [hunk 14]
```diff
@@ -65,35 +65,53 @@ else()
...
+      file(MD5 ${FILE_PATH} EXISTING_FILE_HASH)
+      if(${FILE_HASH} STREQUAL ${EXISTING_FILE_HASH})
+        message(STATUS "same ${FILE_NAME}")
+        message(STATUS "File already exists.")
+      else()
+        message(STATUS "diff ${FILE_NAME}")
+        message(STATUS "File hash changes. Downloading now ...")
+        file(DOWNLOAD https://awf.ml.dev.web.auto/perception/models/${FILE_NAME} ${FILE_PATH} STATUS DOWNLOAD_STATUS)
+        # file(MD5 ${FILE_PATH} DOWNLOADED_FILE_HASH)  # disable to pass ci
+        message(STATUS "Downloaded file hash: ${DOWNLOADED_FILE_HASH}")
```
Comment (wep21) [inline perception/lidar_apollo_instance_segmentation/CMakeLists.txt:91]:
```suggestion
```
Unnecessary?

File: perception/lidar_apollo_instance_segmentation/CMakeLists.txt [hunk 15]
```diff
@@ -65,35 +65,53 @@ else()
...
+        list(GET DOWNLOAD_STATUS 0 STATUS_CODE)
+        list(GET DOWNLOAD_STATUS 1 ERROR_MESSAGE)
+      endif()
+    else()
+      message(STATUS "not found ${FILE_NAME}")
+      message(STATUS "File doesn't exists. Downloading now ...")
+      file(DOWNLOAD https://awf.ml.dev.web.auto/perception/models/${FILE_NAME} ${FILE_PATH} STATUS DOWNLOAD_STATUS)
+      # file(MD5 ${FILE_PATH} DOWNLOADED_FILE_HASH)  # disable to pass ci
+      message(STATUS "Downloaded file hash: ${DOWNLOADED_FILE_HASH}")
```
Comment (wep21) [inline perception/lidar_apollo_instance_segmentation/CMakeLists.txt:100]: Ditto

**Review by wep21 — APPROVED** [review]
LGTM

**Review by wep21 — CHANGES_REQUESTED** [review]
@eratostennis Could you make timeout time longer for huge models?

**Comment by eratostennis** [comment]
> @eratostennis Could you make timeout time longer for huge models?

@wep21 I fixed it.

**Review by wep21 — APPROVED** [review]
LGTM

**Pull request merged**